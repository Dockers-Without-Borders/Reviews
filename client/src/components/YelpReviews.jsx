import React from 'react';
import axios from 'axios';
import Search from './Search.jsx';
import ReviewEntry from './ReviewList.jsx';


class YelpReviews extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      restName: 'RM 212',
      reviewsInfo: []
    }
  }

  //name should be passed down as props, now just for testing
  //get returns an array of objects
  componentDidMount() {
    let name = this.state.restName;
    let url = '/restaurantReviews/' + name;
    axios.get(url)
    .then(data => {
      this.setState({
        reviewsInfo: data
      });
    })
    .catch(err => console.log('error retrieving data', err))
  }




  render() {
    return (
    <div className='reviews_container'>
      <Search />
      <ReviewEntry info={this.state.reviewsInfo}/>
    </div>
    )}
}

export default YelpReviews;