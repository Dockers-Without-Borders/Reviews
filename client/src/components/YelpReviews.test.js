import React from 'react';
import { shallow } from 'enzyme';

import YelpReviews from './YelpReviews';

describe('<YelpReviews />', () => {
  it('Should render component YelpReviews', () => {
    const wrapper = shallow(<YelpReviews />);
    expect(wrapper).toMatchSnapshot();
    
  });
});