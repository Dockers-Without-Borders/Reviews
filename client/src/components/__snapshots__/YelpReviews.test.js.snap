// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YelpReviews /> Should render component YelpReviews 1`] = `ShallowWrapper {}`;
